### ISSUE TITLE
Compiler Should Not Raise CS8618 When Static, Non-Nullable Variable Initialized by Static Constructor in Different File

### ISSUE THREAD

**Issue by justin-millman** [dotnet/roslyn#46121]
**Version Used**:  Visual C# Compiler Version 3.6.0-4.20251.5 (910223b6), used with Visual Studio 2019 (v16.6.2) and C#8 with nullable contexts enabled

**Steps to Reproduce**:
```cs
// File A
namespace Example {
    public partial class Foo {
        private static readonly string SOME_STRING;
    }
}

// File B
namespace Example {
    public partial class Foo {
        static Foo() {
            SOME_STRING = "something";
        }
    }
}
```
The above files are in the same C# .NET CORE 3.1 class library project, in the same directory. I haven't tested to see what happens if they're in different directories, but I can't imagine that it would result in a different behavior.

**Expected Behavior**:
Compiler does not raise CS8618 for `SOME_STRING` because it is, in fact, initialized to a non-null value by the static constructor Foo.

**Actual Behavior**:
Compiler raises CS8618 for `SOME_STRING`
